Failures when build with msvc + permissive- on windows

This issue found when build ITK with permissive- by msvc on windows, could you help have a look?
The issue in 
https://github.com/vxl/vxl/blob/master/config/cmake/config/vxl_platform_tests.cxx#L103
https://github.com/vxl/vxl/blob/master/config/cmake/config/vxl_platform_tests.cxx#L105

properly fix:
change 
void function(int x, char *ptr = "foo");
to
void function(int x, const char *ptr = "foo");

Change
void function(int x, char *ptr) { ++ ptr[x]; }
to
void function(int x, const char *ptr) { ptr[x]; }

Environment: VS2017 latest revision + windows server 2016
Repro:
1. git clone http://itk.org/ITK.git D:\ITK\src
2. Open a VS 2017 x86 command prompt and browse to D:\ITK
3. set _CL_=/D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING /permissive-
4. mkdir build_x86 && pushd build_x86
cmake -G "Visual Studio 15 2017" -DCMAKE_SYSTEM_VERSION=10.0.16299.0 -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTING=ON -DBUILD_EXAMPLES=OFF ..\src\
5.msbuild /m /p:Configuration=Release;Platform=Win32 All_BUILD.vcxproj /t:Rebuild

failures:
D:\ITK\src\Modules\ThirdParty\VNL\src\vxl\config\cmake\config\vxl_platform_tests.cxx(103): error C2440: 'default argument': cannot convert from 'const char [4]' to 'char *' 
D:\ITK\src\Modules\ThirdParty\VNL\src\vxl\config\cmake\config\vxl_platform_tests.cxx(105): error C2440: 'default argument': cannot convert from 'const char [4]' to 'char *'

--- a/config/cmake/config/vxl_platform_tests.cxx
+++ b/config/cmake/config/vxl_platform_tests.cxx
@@ -100,10 +100,10 @@ int main() { return 0; }
 // VCL_DEFAULT_VALUE(x) will be set to "= x" if this test fails, to "" otherwise
 
 // declaration
-void function(int x, char *ptr = "foo");
+char function(int x, const char *ptr = "foo");
 
 // definition
-void function(int x, char *ptr) { ++ ptr[x]; }
+char function(int x, const char *ptr) { return ptr[x]; }
 
 int main() { return 0; }
 #endif // VCL_DEFAULT_VALUE

